### ISSUE TITLE
no-construct-in-interface / no-construct-in-public-property-of-construct: eslint plugin misses Function, Topic, Queue and DatabaseCluster

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#572]
### Describe the bug

`isClassType()` in the eslint plugin tests the symbol flags with strict equality:

```ts
export const isClassType = (type: Type): boolean => {
  return getSymbol(type)?.flags === SymbolFlags.Class;
};
```

`SymbolFlags` is a bit field. Several AWS CDK base classes are declaration-merged with an
interface of the same name via the generated `*-augmentations.generated.d.ts` files
(`declare module "./topic-base" { interface TopicBase { ... } }`), so their symbol flags are
`Transient | Interface | Class` (`33554528`), not `Class` (`32`). The strict comparison fails
and `hasMatchingInterfaceInHierarchy()` stops walking the base chain, so the concrete construct
is never recognised as "a resource with a read-only interface".

The oxlint plugin does not use `isClassType()` at all and detects these types correctly, so
the two plugins disagree.

Affected base classes in `aws-cdk-lib` 2.266.0 (one per augmentation file):
`FunctionBase`, `TopicBase`, `QueueBase`, `DatabaseClusterBase`, `DatabaseInstanceBase`,
`VpnConnectionBase`. In practice this means `lambda.Function`, `sns.Topic`, `sqs.Queue`,
`rds.DatabaseCluster` and `ec2.VpnConnection` are not reported by
`no-construct-in-interface` or `no-construct-in-public-property-of-construct`.

**Reproduction**

`src/fixture.ts`:

```ts
import { Function as LambdaFunction } from "aws-cdk-lib/aws-lambda";
import { DatabaseCluster } from "aws-cdk-lib/aws-rds";
import { Bucket } from "aws-cdk-lib/aws-s3";
import { Topic } from "aws-cdk-lib/aws-sns";
import { Queue } from "aws-cdk-lib/aws-sqs";

export interface MyConstructProps {
  readonly bucket: Bucket;
  readonly fn: LambdaFunction;
  readonly topic: Topic;
  readonly queue: Queue;
  readonly dbCluster: DatabaseCluster;
}
```

`eslint.config.mjs`:

```js
import cdkPlugin from "eslint-plugin-awscdk";
import tseslint from "typescript-eslint";

export default [
  {
    files: ["**/*.ts"],
    plugins: { awscdk: cdkPlugin },
    languageOptions: {
      parser: tseslint.parser,
      parserOptions: { projectService: true, tsconfigRootDir: import.meta.dirname },
    },
    rules: { "awscdk/no-construct-in-interface": "error" },
  },
];
```

`.oxlintrc.json`:

```json
{
  "jsPlugins": ["oxlint-plugin-awscdk"],
  "options": { "typeAware": true },
  "rules": { "awscdk/no-construct-in-interface": "error" }
}
```

Commands and output:

```
$ npx eslint --config eslint.config.mjs src/fixture.ts

/path/to/repro/src/fixture.ts
  8:3  error  Interface property 'bucket' should not use CDK Construct type 'Bucket'. Consider using an interface or type alias instead  awscdk/no-construct-in-interface

✖ 1 problem (1 error, 0 warnings)

$ npx oxlint --format=unix -c .oxlintrc.json src/fixture.ts
src/fixture.ts:8:3: Interface property 'bucket' should not use CDK Construct type 'Bucket'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-interface)]
src/fixture.ts:9:3: Interface property 'fn' should not use CDK Construct type 'Function'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-interface)]
src/fixture.ts:10:3: Interface property 'topic' should not use CDK Construct type 'Topic'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-interface)]
src/fixture.ts:11:3: Interface property 'queue' should not use CDK Construct type 'Queue'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-interface)]
src/fixture.ts:12:3: Interface property 'dbCluster' should not use CDK Construct type 'DatabaseCluster'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-interface)]

5 problems
```

The same divergence appears with `no-construct-in-public-property-of-construct` on
`public readonly topic: Topic = ...` inside a Construct class.

### Expected Behavior

The eslint plugin should report all five properties, matching the oxlint plugin. `Topic`,
`Queue`, `Function` and `DatabaseCluster` all have read-only interfaces (`ITopic`, `IQueue`,
`IFunction`, `IDatabaseCluster`) and are exactly what the rule is meant to catch.

### Current Behavior

Only `Bucket` (and other resources whose base class is not declaration-merged) is reported by
the eslint plugin.

### Plugin

eslint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.1

### Language Version

TypeScript 6.0.3

### Additional Context

```yaml
# machine-readable summary
rules: ["no-construct-in-interface", "no-construct-in-public-property-of-construct"]
symptom: ["false-negative", "cross-plugin-divergence"]
affected_packages: ["eslint-plugin-awscdk"]
root_cause: "packages/eslint/src/core/ts-type/checker/is-class.ts:5"
verified_on: "main @ 6044b32 (eslint-plugin-awscdk 4.4.1, oxlint-plugin-awscdk 1.0.1)"
related_issues: ["#537", "#381"]
```

Compared against oxlint-plugin-awscdk 1.0.1 on the same file; the oxlint plugin reports all five properties.

Root cause verified with the TypeScript compiler API on the same fixture — `flags` is printed
for each type in the base chain:

```
name=Topic     flags=32       isClass=true   decls=ClassDeclaration
  baseTypes=TopicBase
    name=TopicBase  flags=33554528  isClass=false  decls=ClassDeclaration,InterfaceDeclaration
      heritage=extends:Resource ; implements:ITopic|IEncryptedResource
name=Bucket    flags=32       isClass=true   decls=ClassDeclaration
  baseTypes=BucketBase
    name=BucketBase flags=32        isClass=true   decls=ClassDeclaration
      heritage=extends:Resource ; implements:IBucket|IEncryptedResource
```

`33554528` = `Transient (33554432) | Interface (64) | Class (32)`, so
`flags === SymbolFlags.Class` is `false` for `TopicBase` even though it is a class.

Source:

- `packages/eslint/src/core/ts-type/checker/is-class.ts:5` (the `===` comparison)
- `packages/eslint/src/core/cdk-construct/type-checker/is-resource-with-readonly-interface.ts:74`
  (`baseTypes.some((baseType) => isClassType(baseType) && checkTypeAndBases(baseType))`)
- `packages/eslint/src/core/cdk-construct/type-finder/index.ts:12`

A bitmask test — `((getSymbol(type)?.flags ?? 0) & SymbolFlags.Class) !== 0` — fixes it.
(`rds.DatabaseInstance` happens to be reported today only because the concrete class itself
carries `implements IDatabaseInstance`, so the walk never has to descend past the merged base.)

Related prior issues: the detection chain for this rule family has been widened repeatedly
(#198, #201, #216, #218, #220, #222, and most recently **#537** for non-first generic type
arguments) — always by adding another *type shape* to walk. The `isClassType` gate that decides
whether the walk descends into a base class at all was never examined.

**#381** (open) proposes widening the same predicate to `I<Class>Ref` interfaces and
`CfnResource` subclasses and touches the same files. Landing this one-line bitmask fix first
keeps both diffs reviewable; landing #381 without it would leave `Function` / `Topic` / `Queue`
undetected on the eslint side regardless.

Note for regression tests: use `DatabaseCluster`, not `DatabaseInstance` — the latter is
reported today only because the concrete class carries `implements IDatabaseInstance` itself, so
the walk never reaches the merged base.

Reproduction environment:

- macOS (darwin 23.1.0), Node.js v24.18.0, pnpm 10.29.3
- aws-cdk-lib 2.266.0 / constructs 10.8.1
- ESLint 10.8.1 + typescript-eslint 8.67.0 + TypeScript 6.0.3
- oxlint 1.79.0 + oxlint-tsgolint 7.0.2001 (corsa-oxlint 1.12.2)
